### PR TITLE
Apiusagemonitoring context

### PR DESCRIPTION
**Pull request by zeitlinger**
- proper support for API usage monitoring tag
- support for SLO monitoring for API usage monitoring

### CODE REVIEW

File: filters/apiusagemonitoring/spec.go [hunk 1]
```diff
@@ -32,6 +35,7 @@ func NewApiUsageMonitoring(
 	realmKeys string,
 	clientKeys string,
 	realmsTrackingPattern string,
+	tracer opentracing.Tracer,
```
Comment (aryszka) [inline filters/apiusagemonitoring/spec.go]: We could keep backwards compatibility, if we created a new function, e.g. `NewApiUsageMonitoringWithTracing`, and the current one called the new with the NopTracer. Even if there's a low chance that somebody uses this function directly, it's nicer not to assume that nobody does.

**Comment by szuecs** [comment]
:-1: 

I am not really convinced that it makes sense to have opentracing integration in a filter that does not create costs, like call to external services. Therefore I would vote against following this idea. Spans are not free and we already have start stop logs for all filters.

**Comment by szuecs** [comment]
:-1:

**Comment by szuecs** [comment]
We talked internally about this and the decision with @zeitlinger was that there is a simpler way to do this, which will be followed up on. In case this needs to be done like this for whatever reason please reopen this PR.

**Pull request closed without merging**